android/src/main/java/com/onegini/mobile/sdk/reactnative/clean/use_cases/ResourceRequestUseCase.kt

```diff
@@ -51,10 +53,25 @@ class ResourceRequestUseCase @Inject constructor(
     }
 
     private fun buildRequest(requestDetails: ResourceRequestDetails): Request {
-        return Request.Builder()
+        val request =  Request.Builder()
             .url(requestDetails.path)
             .headers(requestDetails.headers)
-            .build()
+        return when (requestDetails.method) {
+            ApiCall.GET -> {
+                request.get()
+            }
+            ApiCall.POST -> {
+                val body = requestDetails.body ?: ""
+                request.post(body.toRequestBody(null))
+            }
+            ApiCall.PUT -> {
+                val body = requestDetails.body ?: ""
+                request.put(body.toRequestBody(null))
+            }
+            ApiCall.DELETE -> {
+                request.delete(requestDetails.body?.toRequestBody())
+            }
+        }.build()
```

Comment:
you could use extension fuction here to improve readablity. Something like this:
```
private fun buildRequest(requestDetails: ResourceRequestDetails): Request {
        return Request.Builder()
            .url(requestDetails.path)
            .headers(requestDetails.headers)
            .setMethod(requestDetails)
            .build()
    }

    private fun Request.Builder.setMethod(requestDetails: ResourceRequestDetails): Request.Builder {
        return when(requestDetails.method){
            ApiCall.GET -> {
                this.get()
            }
            ApiCall.POST -> {
                val body = requestDetails.body ?: ""
                this.post(body.toRequestBody(null))
            }
            ApiCall.PUT -> {
                val body = requestDetails.body ?: ""
                this.put(body.toRequestBody(null))
            }
            ApiCall.DELETE -> {
                this.delete(requestDetails.body?.toRequestBody())
            }
        }
    }
```